jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/CaseServiceImplTest.java

```diff
@@ -3730,4 +3735,48 @@ public void testCaseWithStageAndBoundaryTimerFired() throws InterruptedException
             }
         }
     }
+
+    @Test
+    public void testCaseWithBoundaryTimerFiredAtStage() throws InterruptedException {
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        PrintStream originalOut = System.out;
+
+        System.setOut(new PrintStream(outContent));
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), "CaseWithBoundaryTimerStage");
+        assertNotNull(caseId);
+        assertEquals(FIRST_CASE_ID, caseId);
+        
+        Thread.sleep(3000);
+        
+        try {
+            Collection<CaseStageInstance> stages = caseRuntimeDataService.getCaseInstanceStages(caseId, false, null);
+            assertThat(stages).isNotNull().hasSize(2);
+            Iterator<CaseStageInstance> iterator = stages.iterator();
+
+            CaseStageInstance stage1 = iterator.next();
+            assertThat(stage1.getName()).isEqualTo("Stage 1");
+            assertThat(stage1.getStatus()).isEqualTo(StageStatus.Completed);
+
+            CaseStageInstance stage2 = iterator.next();
+            assertThat(stage2.getName()).isEqualTo("Stage 2");
+            assertThat(stage2.getStatus()).isEqualTo(StageStatus.Active);
+            
+            assertThat(outContent.toString(), containsString("timer on stage"));
+            assertThat(outContent.toString(), containsString("timer on task"));
```

Comment:
I don't think we should reply on this, as mentioned above, using event listeners we can ensure that the timer (based on node info) has been triggered so no need to use the captured system out.

Reply:
Ok, thanks, fixed with the same PR at Enrique's fork